src/Swashbuckle.AspNetCore.Swagger/SwaggerMiddleware.cs

```diff
@@ -33,10 +27,16 @@ public class SwaggerMiddleware
             _next = next;
             _options = options ?? new SwaggerOptions();
             _requestMatcher = new TemplateMatcher(TemplateParser.Parse(_options.RouteTemplate), new RouteValueDictionary());
-
             _swaggerDocumentSerializer = serviceProvider?.GetService<ISwaggerDocumentSerializer>();
         }
 
+        [Obsolete("Remove in next major release, kept for binairy compatibility")]
```

Comment:
Just add this as a `// TODO` comment. Otherwise things that use `TreatWarningsAsErrors=true` will break.

I've added a comment to track this here: https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2798#issuecomment-2060853358